### PR TITLE
feat: add feature to check if hex is an account

### PR DESCRIPTION
**Pull request by maschad**
# Release notes

In this release, we:

- Added Provider methods `isUserAccount` and `getAddressType` which indicate the type of the hex passed

# Summary

We now have two methods on the `Provider` to check the account type

# Checklist

- [x] All **changes** are **covered** by **tests** (or not applicable)
- [x] All **changes** are **documented** (or not applicable)
- [x] I **reviewed** the **entire PR** myself (preferably, on GH UI)
- [x] I **described** all **Breaking Changes** (or there's none)

### CODE REVIEW

File: packages/account/src/providers/provider.ts [hunk 1]
```diff
@@ -1950,6 +1950,53 @@ Supported fuel-core version: ${supportedVersion}.`
...
+        transactionId: id,
+      });
+
+      if (contract || blob || transaction) {
+        return false;
+      }
+      return true;
+    } catch (error) {
+      return true;
+    }
```
Comment (Torres-ssf) [inline packages/account/src/providers/provider.ts:1973]: @Why do we need this validation here? I wouldn't make it possible to false positives?

Comment (maschad) [reply packages/account/src/providers/provider.ts]: Good catch 8bb3be33f

**Review by Torres-ssf — APPROVED** [review]

File: .github/workflows/pr-release.yaml [hunk 2]
```diff
@@ -8,7 +8,7 @@ jobs:
     name: "Release PR to npm"
     runs-on: ubuntu-latest
     # comment out if:false to enable release PR to npm
-    if: false
+    # if: false
```
Comment (Torres-ssf) [inline .github/workflows/pr-release.yaml]: We need to re-comment this line

Comment (petertonysmith94) [inline .github/workflows/pr-release.yaml:11]:
```suggestion
    if: false
```

**Review by Torres-ssf — APPROVED** [review]

File: .github/workflows/pr-release.yaml (hunk 2 above)
Comment (Torres-ssf) [reply .github/workflows/pr-release.yaml]: My bad it was removed already

**Review by arboleya — APPROVED** [review]

**Review by petertonysmith94 — APPROVED** [review]

**Pull request merged**